### PR TITLE
 Add state tag configuration setting

### PR DESCRIPTION
**Pull request by jmassara**
The tag-key used to determine the state of an AMI can now be set with the `AMIQUERY_STATE_TAG` environment variable. The default value if unspecified is "state".

The aliased query parameter "status" still exists and "state" has been added, as well.

In addition, the `tag` query parameter now properly accepts tag values with colons, e.g. `tag=foo:bar:baz` translates to `foo=bar:baz`.

/cc @moofish32 @thomaswhitcomb @ellery44 @jk-intuit

### CODE REVIEW

File: api/query/params.go [hunk 1]
```diff
@@ -39,14 +37,14 @@ func (p *Params) Decode(u *url.URL) error {
 		switch key {
 		case "tag":
 			for _, value := range values {
-				tag := strings.Split(value, ":")
-				if len(tag) != 2 {
+				if i := strings.Index(value, ":"); i != -1 {
```
Comment (eaddingtonwhite) [inline api/query/params.go:40]: @jmassara can we make this simpler w/
 `if strings.Index(value, ":") != -1 {`

File: api/query/params.go [hunk 2]
```diff
@@ -39,14 +37,14 @@ func (p *Params) Decode(u *url.URL) error {
...
+					p.tags[value[:i]] = append(p.tags[value[:i]], value[i+1:])
+				} else {
 					return fmt.Errorf("invalid query tag value: %s", value)
 				}
-				p.tags[tag[0]] = append(p.tags[tag[0]], tag[1])
 			}
-		case amicache.StateTag:
-			p.tags[amicache.StateTag] = append(p.tags[amicache.StateTag], values...)
+		case stateTag, "state", "status": // aliases for the state tag
+			p.tags[stateTag] = append(p.tags[stateTag], values...)
```
Comment (eaddingtonwhite) [inline api/query/params.go:47]: @jmassara is status just so we are backwards compatible?

Comment (jmassara) [reply api/query/params.go:47]: Yes, it was previously hard-coded to `status`.

File: api/query/params.go (hunk 1 above)
Comment (jmassara) [reply api/query/params.go:40]: I need the index value in order to slice the tag.

**Comment by eaddingtonwhite** [comment]
@jmassara Looks good to me. 👍

**Review by jk-intuit — APPROVED** [review]

**Pull request merged**